src/core/MOM_open_boundary.F90

```diff
@@ -3869,11 +3875,13 @@ subroutine update_OBC_segment_data(G, GV, US, OBC, tv, h, Time)
   if (OBC%add_tide_constituents) time_delta = US%s_to_T * time_type_to_real(Time - OBC%time_ref)
 
   if (OBC%remap_answer_date >= 20190101) then
-    h_neglect = GV%H_subroundoff ; h_neglect_edge = GV%H_subroundoff
+    dz_neglect = GV%dZ_subroundoff ; dz_neglect_edge = GV%dZ_subroundoff
   elseif (GV%Boussinesq) then
-    h_neglect = GV%m_to_H * 1.0e-30 ; h_neglect_edge = GV%m_to_H * 1.0e-10
+    dz_neglect = US%m_to_Z * 1.0e-30 ; dz_neglect_edge = US%m_to_Z * 1.0e-10
+  elseif (GV%semi_Boussinesq) then
+    dz_neglect = GV%kg_m2_to_H*GV%H_to_Z * 1.0e-30 ; dz_neglect_edge = GV%kg_m2_to_H*GV%H_to_Z * 1.0e-10
   else
-    h_neglect = GV%kg_m2_to_H * 1.0e-30 ; h_neglect_edge = GV%kg_m2_to_H * 1.0e-10
+    dz_neglect = GV%dZ_subroundoff ; dz_neglect_edge = GV%dZ_subroundoff
   endif
```

Comment:
If you wanted to avoid duplicating the first and last case, then I think this will work
```
if (OBC%remap_answer_date < 20190101 .and. GV%Boussinesq) then
   dz_neglect = US%m_to_Z * 1.0e-30 ; dz_neglect_edge = US%m_to_Z * 1.0e-10
elseif (OBC%remap_answer_date < 20190101 .and. GV%semi_Boussinesq) then
   dz_neglect = GV%kg_m2_to_H*GV%H_to_Z * 1.0e-30 ; dz_neglect_edge = GV%kg_m2_to_H*GV%H_to_Z * 1.0e-10
else
   dz_neglect = GV%dZ_subroundoff ; dz_neglect_edge = GV%dZ_subroundoff
endif
```

Reply:
Thank you for this suggestion, @marshallward .  The PR has been modified accordingly.